server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -124,8 +128,10 @@ func (s *BuildBuddyServer) GetGroup(ctx context.Context, req *grpb.GetGroupReque
 	if userDB == nil {
 		return nil, status.UnimplementedError("Not Implemented")
 	}
-	group := &tables.Group{
-		URLIdentifier: req.GetUrlIdentifier(),
+	group := &tables.Group{}
+	urlIdentifier := strings.TrimSpace(req.GetUrlIdentifier())
+	if urlIdentifier != "" {
+		group.URLIdentifier = &urlIdentifier
 	}
 	if err := userDB.FillGroup(ctx, group); err != nil {
```

Comment:
Took me a minute to grok this -- I might be a dummy but FillGroup is -- taking input and output in the same argument which is error prone.

Let's remove FillGroup and replace it with GetGroupByID and/or GetGroupByURLIdentifier?

Reply:
Sure, will do that in a separate PR since it's unrelated. Don't want to block this PR on it

Reply:
SGTM.